src/components/form/form.js

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import SingleChoice from "./singleChoice";
+import "./form.scss";
+import { Button } from "../button/button";
+import Results from "./results";
+import _ from "underscore";
+
+const FIRST_NODE = 1;
+
+function Form({ data, evaluateForm, onFinishingForm }) {
+  const [formState, setFormState] = useState({});
+  const [currentNode, setCurrentNode] = useState(data.firstNodeId);
+  const [historyOfSteps, setHistoryOfSteps] = useState([]);
+
+  const resetForm = () => {
+    setCurrentNode(data.firstNodeId);
+    setFormState({});
+    setHistoryOfSteps([]);
+  };
+
+  const answerCurrentQuestion = answer => {
+    setFormState({
+      ...formState,
+      [answer.questionId]: answer.value
+    });
+  };
+
+  const toMap = form => {
+    const groupedById = _.groupBy(form, question => question.questionId);
+    return _.mapObject(groupedById, listOfQuestions => listOfQuestions[0]);
+  };
+
+  const formAsMap = toMap(data.form);
+
+  const createFormWithAnswers = () => {
+    const answersById = _.mapObject(formAsMap, (question, id) => {
+      return {
+        id: question.questionId,
+        questionText: question.questionText,
+        answer: formState[id]
+      };
+    });
+    return {
+      formId: data.formId,
+      timestamp: Date.now(),
+      answers: _.values(answersById)
+    };
+  };
+
+  const questionView = () => {
+    const currentQuestion = formAsMap[currentNode];
+    // TODO: add components for other question types
+    switch (currentQuestion.type) {
+      case "SINGLE_CHOICE": {
+        return (
+          <SingleChoice
+            question={currentQuestion}
+            currentResponse={formState[currentQuestion.questionId]}
+            onAnswer={answerCurrentQuestion}
+          />
+        );
+      }
+      case "FINAL": {
+        onFinishingForm(createFormWithAnswers());
+        return (
+          <Results
+            option={evaluateForm(formState, currentQuestion.options)}
+            question={currentQuestion}
+          />
+        );
+      }
+      default:
+        return null;
+    }
+  };
+
+  const getNextQuestionForOptionValue = (options, optionValue) => {
+    const selectedOption = _.find(
+      options,
+      option => option.value === optionValue
+    );
+    return selectedOption.nextQuestionId;
+  };
+  const goToNextQuestion = () => {
+    // TODO use the disabled prop once the Button component implements it
```

Comment:
We have [disabled](https://taskforce-fe-compone-git-fork-mmircea16-form-with-decisi-2be921.code4ro.now.sh/?path=/story/button--disabled) state for the button now